cert_manager/_helpers.py

```diff
@@ -10,7 +10,12 @@
     from urllib.parse import unquote  # pylint: disable=unused-import
 
 from requests.exceptions import HTTPError
+from requests import PreparedRequest
 
+def prepare_url(url, params={}):
```

Comment:
Alright. Looking back the utility function is not necessary after all as there is indeed a more straightforward approach.

I think the docstring you pointed to cautions against _sending_ a manually instantiated `PreparedRequest` (I only used it to get the URL plus query string) rather than implying it is a private API -- both `Request.prepare()` and `Session.prepare_request()` work similarly so it seems safe.